assets/js/pages/ActivityLogPage/ActivityLogPage.jsx

```diff
@@ -103,19 +156,21 @@ function ActivityLogPage() {
   )(searchParams);
 
   const fetchActivityLog = () => {
-    setLoading(true);
-
-    activityLogRequest(searchParams)
-      .then(({ data }) => {
-        setActivityLogResponse({
-          data: data?.data || [],
-          pagination: data?.pagination,
-        });
-      })
-      .catch(() => setActivityLogResponse(emptyResponse))
-      .finally(() => {
-        setLoading(false);
-      });
+    // defer the loading state to avoid flickering
+    const tid = setTimeout(() => setActivityLogRequest(request.loading()), 500);
```

Comment:
**question:** how can the flickering be reproduced?

Reply:
Good question, it is something visually annoying that has no functional effect. Maybe it only applies to local executions, when the latency is near zero.

Reply:
Alright thanks for clarifying.